packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,222 @@
+import { attr, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { keyEscape } from '@microsoft/fast-web-utilities';
+import { DrawerType, DrawerPosition, DrawerSize } from './drawer.options.js';
+
+/**
+ * Drawer
+ *
+ * A Drawer component for creating modal or non-modal drawers with various configurations.
+ * @extends FASTElement
+ *
+ * @attr {DrawerType} type - Determines whether the drawer should be displayed as modal, non-modal, or alert. When in modal or alert mode, an overlay is applied over the rest of the view.
+ * @attr {boolean} inline - Sets the drawer as an inline element.
+ * @attr {DrawerPosition} position - Sets the position of the drawer (start/end).
+ * @attr {DrawerSize} size - Sets the size of the drawer (small/medium/large).
+ * @attr {string} aria-labelledby - The ID of the element that labels the drawer.
+ * @attr {string} aria-describedby - The ID of the element that describes the drawer.
+ *
+ * @csspart dialog - The dialog element that represents the drawer.
+ *
+ * @slot - Default slot for the drawer's content.
+ *
+ * @fires toggle - Emitted after the drawer's open state changes
+ * @fires beforeToggle - Emitted before the drawer's open state changes
+ *
+ * @method connectedCallback() - Called when the custom element is connected to the document's DOM.
+ * @method show() - Opens the drawer if it is currently hidden.
+ * @method hide() - Closes the drawer if it is currently open.
+ * @method clickHandler(event) - Handles click events on the drawer.
+ * @method keydownHandler(event) - Handles keydown events on the drawer.
+ * @method typeChanged(oldValue, newValue) - Handles changes to the `type` attribute.
+ * @method inlineChanged(oldValue, newValue) - Handles changes to the `inline` attribute.
+ *
+ * @summary A flexible drawer component that can be used in various configurations such as modal, non-modal, alert, inline, overlay, with different sizes and positions.
+ *
+ * @tag fluent-drawer
+ */
+export class Drawer extends FASTElement {
+  /**
+   * @public
+   * The connectedCallback method of the custom element
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+    Updates.enqueue(() => {
+      this.validateConfiguration();
+    });
+  }
+
+  /**
+   * @public
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   */
+  @attr({ attribute: 'type' })
```

Comment:
If it's one word you don't need the properties.

```suggestion
  @attr
```